src/assets.zig

```diff
@@ -42,76 +85,116 @@ fn readDefaultFile(allocator: NeverFailingAllocator, dir: std.fs.Dir) !ZonElemen
 	return .null;
 }
 
-/// Reads .zig.zon files recursively from all subfolders.
-pub fn readAllZonFilesInAddons(externalAllocator: NeverFailingAllocator, addons: main.List(std.fs.Dir), addonNames: main.List([]const u8), subPath: []const u8, defaults: bool, output: *std.StringHashMap(ZonElement)) void {
+/// Reads all asset `.zig.zon` files recursively from all sub folders.
+///
+/// Files red are stored in output hashmap with asset ID as key.
+/// Asset ID are constructed as `{addonName}:{relativePathNoSuffix}`.
+/// relativePathNoSuffix is always unix style path with all extensions removed.
+pub fn readAllZonFilesInAddons(
+	externalAllocator: NeverFailingAllocator,
+	addons: main.List(std.fs.Dir),
+	addonNames: main.List([]const u8),
+	assetSubdirectory: []const u8,
+	defaults: bool,
+	output: *std.StringHashMap(ZonElement),
+	migrations: ?*std.StringHashMap(ZonElement),
+) void {
 	for(addons.items, addonNames.items) |addon, addonName| {
-		var dir = addon.openDir(subPath, .{.iterate = true}) catch |err| {
-			if(err != error.FileNotFound) {
-				std.log.err("Could not open addon directory {s}: {s}", .{subPath, @errorName(err)});
+		var addonAssetsSubdirectory = addon.openDir(assetSubdirectory, .{.iterate = true}) catch |e| {
+			if(e != error.FileNotFound) {
+				std.log.err("Could not open addon directory {s}: {s}", .{assetSubdirectory, @errorName(e)});
 			}
 			continue;
 		};
-		defer dir.close();
+		defer addonAssetsSubdirectory.close();
 
 		var defaultsArena: main.utils.NeverFailingArenaAllocator = .init(main.stackAllocator);
 		defer defaultsArena.deinit();
 
-		const defaultsArenaAllocator = defaultsArena.allocator();
-
-		var defaultMap = std.StringHashMap(ZonElement).init(defaultsArenaAllocator.allocator);
-
-		var walker = dir.walk(main.stackAllocator.allocator) catch unreachable;
-		defer walker.deinit();
+		const defaultsAllocator = defaultsArena.allocator();
+		var defaultMap = std.StringHashMap(ZonElement).init(defaultsAllocator.allocator);
+
+		var directoryWalker = addonAssetsSubdirectory.walk(main.stackAllocator.allocator) catch unreachable;
+		defer directoryWalker.deinit();
+
+		while(directoryWalker.next() catch |e| block_break: {
+			std.log.err("Got error while iterating addon directory {s}: {s}", .{assetSubdirectory, @errorName(e)});
+			break :block_break null;
+		}) |directoryEntry| {
+			if(
+				(directoryEntry.kind != .file)
+				or (std.ascii.startsWithIgnoreCase(directoryEntry.basename, "_defaults"))
+				or (!std.ascii.endsWithIgnoreCase(directoryEntry.basename, ".zon"))
+				or (std.ascii.startsWithIgnoreCase(directoryEntry.path, "textures"))
+			){
+				continue;
+			}
```

Comment:
Please revert to the old version of the if, all the indentation change make it impossible to see what you actually changed in the following section.